hypothesis-python/tests/conjecture/test_alt_backend.py

```diff
@@ -467,6 +467,26 @@ def test_function(data):
         assert "Draw 1: <symbolic>" in out.getvalue()
 
 
+def test_realization_with_observability():
+    with temp_register_backend("realize", RealizeProvider):
+
+        @given(st.data())
+        @settings(backend="realize")
+        def test_function(data):
+            data.draw(st.integers())
+
+        with capture_observations() as observations:
+            test_function()
+
+    test_cases = [tc for tc in observations if tc["type"] == "test_case"]
+    assert {tc["representation"] for tc in test_cases} == {
+        # from the first ChoiceTemplate(type="simplest") example
+        "test_function(\n    data=data(...),\n)\nDraw 1: 0",
+        # from all other examples. data=<symbolic> probably isn't ideal.
+        "test_function(\n    data=<symbolic>,\n)\nDraw 1: 42",
+    }
```

Comment:
I think the draws should be `<symbolic>` right?

And agreed, it might be worth special-casing `DataObject` in this pprint logic... no, we can't do that.  Check if the strategy passed to `@given()` was `== st.data()` instead?

Reply:
we're constructing this representation after the test function finishes, so we can call `provider.realize` on them - that's why it's a constant 42 from the `RealizeProvider` here

Reply:
oh, nice!  code comment to that effect for future-Zac please 🙏

Reply:
(though this won't help if you hold a reference to a symbolic object which you use in `__str__`)

Reply:
> I'll continue to advocate for a `PrimitiveProvider.is_symbolic` method, which would solve this and similar cases!

I didn't think this was feasible in general, and if it's not a complete solution I'd rather do something else 🤔